src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -360,7 +360,14 @@ private void RecreateIndexes(DatabaseRecord databaseRecord)
                             case IndexType.JavaScriptMap:
                             case IndexType.JavaScriptMapReduce:
                                 var indexDefinition = index.GetIndexDefinition();
-                                databaseRecord.Indexes.Add(indexDefinition.Name, indexDefinition);
+                                if (indexDefinition.Name.StartsWith(Constants.Documents.Indexing.SideBySideIndexNamePrefix, StringComparison.OrdinalIgnoreCase))
+                                {
+                                    // the side by side index is the last version of this index
+                                    // and it's the one that should be stored in the database record
+                                    indexDefinition.Name = indexDefinition.Name[Constants.Documents.Indexing.SideBySideIndexNamePrefix.Length..];
+                                }
+
+                                databaseRecord.Indexes[indexDefinition.Name] = indexDefinition;
```

Comment:
Here you relay on the order of `Directory.GetDirectories(indexesPath)`, no? If the original index is returned later then you will override the replacement definition by the old one.

Reply:
Please write a test for that case

Reply:
done